core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -429,74 +429,72 @@ private[spark] class Executor(
+          // Attempt to exit cleanly by informing the driver of our failure.
+          // If anything goes wrong (or this was a fatal exception), we will delegate to
+          // the default uncaught exception handler, which will terminate the Executor.
+          logError(s"Exception in $taskName (TID $taskId)", t)
+
+          // Collect latest accumulator values to report back to the driver
+          val accums: Seq[AccumulatorV2[_, _]] =
+            if (task != null) {
+              task.metrics.setExecutorRunTime(
+                math.max(System.nanoTime() - taskStart, 0L) / 1000000.0)
+              val taskEndCpu = if (threadMXBean.isCurrentThreadCpuTimeSupported) {
+                threadMXBean.getCurrentThreadCpuTime
+              } else 0L
+              task.metrics.setExecutorCpuTime(
+                math.max(taskEndCpu - taskStartCpu, 0L) / 1000000.0)
+              task.metrics.setJvmGCTime(computeTotalGcTime() - startGCTime)
+              task.collectAccumulatorUpdates(taskFailed = true)
+            } else {
+              Seq.empty
+            }
 
-            val accUpdates = accums.map(acc => acc.toInfo(Some(acc.value), None))
+          val accUpdates = accums.map(acc => acc.toInfo(Some(acc.value), None))
 
-            val serializedTaskEndReason = {
-              try {
-                ser.serialize(new ExceptionFailure(t, accUpdates).withAccums(accums))
-              } catch {
-                case _: NotSerializableException =>
-                  // t is not serializable so just send the stacktrace
-                  ser.serialize(new ExceptionFailure(t, accUpdates, false).withAccums(accums))
-              }
+          val serializedTaskEndReason = {
+            try {
+              ser.serialize(new ExceptionFailure(t, accUpdates).withAccums(accums))
+            } catch {
+              case _: NotSerializableException =>
+                // t is not serializable so just send the stacktrace
+                ser.serialize(new ExceptionFailure(t, accUpdates, false).withAccums(accums))
             }
-            setTaskFinishedAndClearInterruptStatus()
-            execBackend.statusUpdate(taskId, TaskState.FAILED, serializedTaskEndReason)
-
-            // Don't forcibly exit unless the exception was inherently fatal, to avoid
-            // stopping other tasks unnecessarily.
-            if (Utils.isFatalError(t)) {
-              if (!isLocal) {
-                Thread.getDefaultUncaughtExceptionHandler.
+          }
+          setTaskFinishedAndClearInterruptStatus()
+          execBackend.statusUpdate(taskId, TaskState.FAILED, serializedTaskEndReason)
+
+          // Don't forcibly exit unless the exception was inherently fatal, to avoid
+          // stopping other tasks unnecessarily.
+          if (Utils.isFatalError(t)) {
+            if (!isLocal) {
+              Thread.getDefaultUncaughtExceptionHandler.
                   uncaughtException(Thread.currentThread(), t)
```

Comment:
Are these changes mostly reversion of the changes done earlier? What does the diff against stock spark's version looks like?